src/compile/data/bin.ts

```diff
@@ -89,6 +90,27 @@ export class BinNode extends DataFlowNode {
     return new BinNode(bins);
   }
 
+  public static makeBinFromTransform(model: Model, t: BinTransform) {
+    const bins: Dict<BinComponent> = {};
+
+    if (t.bin) {
+      const bin: Bin = isBoolean(t.bin) ? {} : t.bin;
+      const key = `${binToString(t.bin)}_${t.field}`;
+      bins[key] = {
+        bin: bin,
+        field: t.field,
+        as: [binToString(t.bin) + '_' + t.field + '_start', binToString(t.bin) + '_' + t.field + '_end'],
```

Comment:
I manually extracted the correct logic from what we have in the other method without encoding channel